docs/src/_guide/troubleshooting.md

```diff
@@ -0,0 +1,87 @@
+
+```bash
+MELTANO_CLI_LOG_LEVEL=debug meltano elt ...
+
+meltano --log-level=debug elt ...
+```
+
+In debug mode, meltano will log the arguments and environment used to invoke the Singer tap and target executables (and dbt, when running transformations), including the paths to the generated config, catalog, and state files, for you to review:
+
+```
+$ meltano --log-level=debug elt tap-gitlab target-jsonl --state-id=gitlab-to-jsonl
+meltano            | INFO Running extract & load...
+meltano            | INFO Found state from 2020-08-05 21:30:20.487312.
+meltano            | DEBUG Invoking: ['demo-project/.meltano/extractors/tap-gitlab/venv/bin/tap-gitlab', '--config', 'demo-project/.meltano/run/tap-gitlab/tap.config.json', '--state', 'demo-project/.meltano/run/tap-gitlab/state.json']
+meltano            | DEBUG Env: {'TAP_GITLAB_API_URL': 'https://gitlab.com', 'GITLAB_API_TOKEN': '', 'GITLAB_API_GROUPS': '', 'GITLAB_API_PROJECTS': 'meltano/meltano', 'GITLAB_API_ULTIMATE_LICENSE': 'False', 'GITLAB_API_START_DATE': '2021-03-01'}
+meltano            | DEBUG Invoking: ['demo-project/.meltano/loaders/target-jsonl/venv/bin/target-jsonl', '--config', 'demo-project/.meltano/run/target-jsonl/target.config.json']
+meltano            | DEBUG Env: {'MELTANO_EXTRACTOR_NAME': 'tap-gitlab', 'MELTANO_EXTRACTOR_NAMESPACE': 'tap_gitlab', 'MELTANO_EXTRACT_API_URL': 'https://gitlab.com', 'MELTANO_EXTRACT_PRIVATE_TOKEN': '', 'MELTANO_EXTRACT_GROUPS': '', 'MELTANO_EXTRACT_PROJECTS': 'meltano/meltano', 'MELTANO_EXTRACT_ULTIMATE_LICENSE': 'False', 'MELTANO_EXTRACT_START_DATE': '2021-03-01', 'TAP_GITLAB_API_URL': 'https://gitlab.com', 'GITLAB_API_TOKEN': '', 'GITLAB_API_GROUPS': '', 'GITLAB_API_PROJECTS': 'meltano/meltano', 'GITLAB_API_ULTIMATE_LICENSE': 'False', 'GITLAB_API_START_DATE': '2021-03-01', 'TARGET_JSONL_DESTINATION_PATH': 'output', 'TARGET_JSONL_DO_TIMESTAMP_FILE': 'False'}
+```
+
+## Dump Files Generated by Running Meltano Commands to STDOUT
+
+The ```--dumps``` option can be passed (along with any of the other options) to dump the content of a pipeline-specific generated file to STDOUT instead of actually running the pipeline.
+
+This can aid in debugging extractor catalog generation, incremental replication state lookup, and pipeline environment variables.
+
+Supported values are:
+
+- **catalog**: Dump the extractor catalog file that would be passed to the tap’s executable using the ```--catalog``` option.
+- **state**: Dump the extractor state file that would be passed to the tap’s executable using the ```--state``` option.
+- **extractor-config**: Dump the extractor config file that would be passed to the tap’s executable using the ```--config``` option.
+- **loader-config**: Dump the loader config file that would be passed to the target’s executable using the ```--config``` option.
+
+Like any standard output, the dumped content can be redirected to a file using >, e.g. ```meltano elt ... --dump=state > state.json```.
```

Comment:
```suggestion
Like any standard output, the dumped content can be redirected to a file using >, e.g. `meltano elt ... --dump=state > state.json`.
```

Reply:
@tayloramurphy - I think the proper way to dump state now is with `meltano state get`...

Reply:
@aaronsteers agreed - can you open a separate issue on that? This is basically a verbatim copy from the CLI docs so I'm ok merging it as is.